osquery/tables/system/windows/users.cpp

```diff
@@ -29,40 +30,48 @@ std::string psidToString(PSID sid);
 int getUidFromSid(PSID sid);
 int getGidFromSid(PSID sid);
 
+const std::string kRegProfilePath =
+    "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows "
+    "NT\\CurrentVersion\\ProfileList";
+const char kRegSep = '\\';
+
 namespace tables {
 
-void ProcessWinProfile(const WmiResultItem& wmiResult,
-                       std::set<std::string>& processedSids,
-                       QueryData& results) {
-  Row r;
-  std::string sidString;
-  wmiResult.GetString("SID", sidString);
-
-  r["uuid"] = sidString;
-  processedSids.insert(sidString);
-  wmiResult.GetString("LocalPath", r["directory"]);
-
-  PSID sid;
-  auto ret = ConvertStringSidToSidA(sidString.c_str(), &sid);
-  if (ret == 0) {
-    VLOG(1) << "Convert SID to string failed with " << GetLastError();
-  }
-  r["uid"] = INTEGER(getUidFromSid(sid));
-  r["gid"] = INTEGER(getGidFromSid(sid));
-  r["uid_signed"] = r["uid"];
-  r["gid_signed"] = r["gid"];
-  r["shell"] = "C:\\Windows\\system32\\cmd.exe";
-
-  WmiRequest accntReq(
-      "select Description, Name from Win32_UserAccount where sid = \"" +
-      r["uuid"] + "\"");
-  auto& accntResults = accntReq.results();
-  if (accntReq.getStatus().ok() && !accntResults.empty()) {
-    accntResults[0].GetString("Description", r["description"]);
-    accntResults[0].GetString("Name", r["username"]);
-  } else {
+void processRoamingProfiles(const std::set<std::string>& processedSids,
+                            QueryData& results) {
+  QueryData regResults;
+  queryKey(kRegProfilePath, regResults);
+
+  for (const auto& profile : regResults) {
+    Row r;
+    if (profile.at("type") != "subkey") {
+      continue;
+    }
+    auto sidString = profile.at("name");
+    r["uuid"] = sidString;
+
+    QueryData profileDetails;
+    queryKey(
+        kRegProfilePath + kRegSep + sidString + kRegSep + "ProfileImagePath",
```

Comment:
I don't think this will return data, since ProfileImagePath is a reg value and not a key. Being able to infer this and do the appropriate reg query in queryKeys is tbd.
```
osquery> select * from registry where key = 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-80-1511743022-2277352901-878227523-1247458543-3245048236\ProfileImagePath';
osquery> select * from registry where key = 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-80-1511743022-2277352901-878227523-1247458543-3245048236' and name = 'ProfileImagePath';
+--------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------+---------------+-------------------------------+------------+
| key
                                                     | path

                                   | name             | type          | data
              | mtime      |
+--------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------+---------------+-------------------------------+------------+
| HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-80-1511743022-2277352901-878227523-1247458543-3245048236 | HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList\S-1-5-80-1511743022-2277352901-878227523-1247458543-3245048236\ProfileImagePath | ProfileImagePath | REG_EXPAND_SZ | C:\Users\HiveStreamingService | 1490043999 |
+--------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------+---------------+-------------------------------+------------+
```
You could do the following, or just execute something like the above SQL to get the data.
```
queryKey(profile.at("path") + kRegSep, profileDetails);
r["LocalPath"] = "";
for (const auto& det: profileDetails) {
  if (det["name"] == "ProfileImagePath") {
    r["LocalPath"] = det["data"];
  }
} 
```